source/extensions/transport_sockets/tls/ssl_socket.cc

```diff
@@ -142,8 +142,14 @@ Network::IoResult SslSocket::doRead(Buffer::Instance& read_buffer) {
         case SSL_ERROR_ZERO_RETURN:
           end_stream = true;
           break;
+        case SSL_ERROR_SYSCALL:
+          if (result.error_.value() == 0) {
+            end_stream = true;
```

Comment:
It's not clear to me from the openssl/boringssl documentation in what cases we'd get `SSL_ERROR_SYSCALL` instead of `SSL_ERROR_ZERO_RETURN`. Can you add some comments about which events each case covers?

Reply:
As @antoniovicente mentioned, it's about graceful shutdown using close_notify alert vs connection closed. I'll add comments.

Reply:
Added, PTAL.